cve_bin_tool/extractor.py

```diff
@@ -74,32 +74,43 @@ def can_extract(self, filename):
                 return True
         return False
 
-    @staticmethod
-    async def extract_file_tar(filename, extraction_path):
+    def tar_member_filter(self, members, extraction_path):
+        """Generator function to serve as a backported filter for tarfile extraction
+        based on https://docs.python.org/3/library/tarfile.html#examples
+        """
+        for tarmember in members:
+            if tarmember.isfile() and str(
+                Path(extraction_path, tarmember.name).resolve()
+            ).startsWith(extraction_path):
+                yield tarmember
+
+    async def extract_file_tar(self, filename, extraction_path):
         """Extract tar files"""
+
+        # make sure we have full path for later checks
+        extraction_path = str(Path(extraction_path).resolve())
         with ErrorHandler(mode=ErrorMode.Ignore) as e:
+            tar = tarfile.open(filename)
             # Python 3.12 has a data filter we can use in extract
             # tarfile has this available in older versions as well
             if hasattr(tarfile, "data_filter"):
-                await aio_unpack_archive(filename, extraction_path, filter="data")
+                tar.extractall(path=extraction_path, filter="data")  # nosec
+                # nosec line because bandit doesn't understand filters yet
+
+            # FIXME: the backported fix is not working on windows.
+            # this leaves the current (unsafe) behaviour so we can fix at least one OS for now
+            if sys.platform == "win32":
```

Comment:
I should probably read the spec, but this code opens the possibility to tarfile having the data_filter attribute *and* being a windows system, in which case you would extract the file twice, right? Maybe the right way to do this is by returning above, right after tar.extractall with return e.exit_code?

Reply:
I think this just needs to be an `elif` (I remember thinking that but apparently didn't actually *write* it... le sigh.)